File: fsharp-giraffe-server/OpenAPI/src/model/Messages.fs
namespace OpenAPI.Model

open System
open System.Collections.Generic
open OpenAPI.Model.AnyType
open OpenAPI.Model.MessagesAllOf
open OpenAPI.Model.MessagesBase

module Messages = 

  //#region Messages


  type Messages = {
    AvatarUrl : AnyType;
    Client : AnyType;
    Content : AnyType;
    ContentType : AnyType;
    DisplayRecipient : AnyType;
    Id : AnyType;
    IsMeMessage : AnyType;
    Reactions : AnyType;
    RecipientId : AnyType;
    SenderEmail : AnyType;
    SenderFullName : AnyType;
    SenderId : AnyType;
    SenderRealmStr : AnyType;
    StreamId : AnyType;
    Subject : AnyType;
    TopicLinks : AnyType;
    Submessages : AnyType;
    Timestamp : AnyType;
    Type : AnyType;
  }
  //#endregion
  

File: scalaz/src/main/scala/org/openapitools/client/api/Messages.scala
package org.openapitools.client.api

import argonaut._
import argonaut.EncodeJson._
import argonaut.DecodeJson._

import org.http4s.{EntityDecoder, EntityEncoder}
import org.http4s.argonaut._
import org.joda.time.DateTime


import Messages._

case class Messages (
  avatarUrl: Option[AnyType],
client: Option[AnyType],
content: Option[AnyType],
`contentType`: Option[AnyType],
displayRecipient: Option[AnyType],
id: Option[AnyType],
isMeMessage: Option[AnyType],
reactions: Option[AnyType],
recipientId: Option[AnyType],
senderEmail: Option[AnyType],
senderFullName: Option[AnyType],
senderId: Option[AnyType],
senderRealmStr: Option[AnyType],
streamId: Option[AnyType],
subject: Option[AnyType],
topicLinks: Option[AnyType],
submessages: Option[AnyType],
timestamp: Option[AnyType],
`type`: Option[AnyType])

object Messages {
  import DateTimeCodecs._

  implicit val MessagesCodecJson: CodecJson[Messages] = CodecJson.derive[Messages]
  implicit val MessagesDecoder: EntityDecoder[Messages] = jsonOf[Messages]
  implicit val MessagesEncoder: EntityEncoder[Messages] = jsonEncoderOf[Messages]
}
